gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala

```diff
@@ -157,14 +158,12 @@ abstract class HashAggregateExecBaseTransformer(
   override protected def outputExpressions: Seq[NamedExpression] = resultExpressions
 
   // Check if Pre-Projection is needed before the Aggregation.
-  protected def needsPreProjection: Boolean = {
-    groupingExpressions.exists {
-      case _: Attribute => false
-      case _ => true
-    } || aggregateExpressions.exists {
+  override def needsPreProjection: Boolean = {
+    needsPreProjection(groupingExpressions) ||
+    aggregateExpressions.exists {
       expr =>
         expr.filter match {
-          case None | Some(_: Attribute) | Some(_: Literal) =>
+          case None | Some(_: Attribute) =>
```

Comment:
There are two reasons for this:
- If the filter condition is a Literal, it can only be of boolean type. Such filters are typically removed during the optimization process in Spark. You can refer to the "EliminateAggregateFilter" function in Spark for more information.
- filter in velox only support `FieldAccessTypedExpr`, but it is possible that CK supports Literal filter.

Reply:
Thanks. Makes sense.